[CI] PointInTimeRelocationIT testPointInTimeRelocationClosingSourceContexts failing

**Build Scans:**
- [elasticsearch-intake #43236 / part2](https://gradle-enterprise.elastic.co/s/w4zwm5uinh7qo)
- [elasticsearch-pull-request #162343 / part-2](https://gradle-enterprise.elastic.co/s/yv25xcafq44qc)
- [elasticsearch-pull-request #162342 / part-2](https://gradle-enterprise.elastic.co/s/uihtofjr7boqi)

**Reproduction Line:**
```
./gradlew ":x-pack:plugin:stateless:internalClusterTest" --tests "org.elasticsearch.xpack.stateless.recovery.PointInTimeRelocationIT.testPointInTimeRelocationClosingSourceContexts" -Dtests.seed=A52CD634CF1535A0 -Dtests.locale=en-GE -Dtests.timezone=Europe/Madrid -Druntime.java=26
```

**Applicable branches:**
main

**Reproduces locally?:**
N/A

**Failure History:**
[See dashboard](https://es-delivery-stats.elastic.dev/app/dashboards#/view/dcec9e60-72ac-11ee-8f39-55975ded9e63?_g=(refreshInterval:(pause:!t,value:60000),time:(from:now-7d%2Fd,to:now))&_a=(controlGroupState:(initialChildControlState:('0c0c9cb8-ccd2-45c6-9b13-96bac4abc542':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:task.keyword,order:0,selectedOptions:!(),title:'GradleTask',type:optionsListControl),'4e6ad9d6-6fdc-4fcc-bf1a-aa6ca79e0850':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:className.keyword,order:1,selectedOptions:!(org.elasticsearch.xpack.stateless.recovery.PointInTimeRelocationIT),title:'Suite',type:optionsListControl),'144933da-5c1b-4257-a969-7f43455a7901':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:name.keyword,order:2,selectedOptions:!('testPointInTimeRelocationClosingSourceContexts'),title:'Test',type:optionsListControl)))))

**Failure Message:**
```
java.lang.AssertionError: pit1 should have changed.
```

**Issue Reasons:**
- [main] 1 consecutive failures in test testPointInTimeRelocationClosingSourceContexts
- [main] 1 consecutive failures in step part2
- [main] 1 consecutive failures in pipeline elasticsearch-intake
- [main] 3 failures in test testPointInTimeRelocationClosingSourceContexts (0.8% fail rate in 354 executions)
- [main] 2 failures in step part-2 (0.9% fail rate in 227 executions)
- [main] 2 failures in pipeline elasticsearch-pull-request (0.9% fail rate in 227 executions)

**Note:**
This issue was created using new test triage automation. Please report issues or feedback to es-delivery.

--- a/muted-tests.yml
+++ b/muted-tests.yml
@@ -769,9 +769,6 @@ tests:
 - class: org.elasticsearch.xpack.esql.ccq.MultiClusterSpecStatsIT
   method: test {csv-spec:stats.sumWithOverflowGroupingRow}
   issue: https://github.com/elastic/elasticsearch/issues/153434
-- class: org.elasticsearch.xpack.stateless.recovery.PointInTimeRelocationIT
-  method: testPointInTimeRelocationClosingSourceContexts
-  issue: https://github.com/elastic/elasticsearch/issues/153452
 - class: org.elasticsearch.xpack.esql.qa.ndjson.NdJsonCompressedFormatSpecIT
   method: test {csv-spec:external-multifile-resolution.ubnSalaryStats [ndjson.zstd/AZURE]}
   issue: https://github.com/elastic/elasticsearch/issues/153470

--- a/x-pack/plugin/stateless/src/internalClusterTest/java/org/elasticsearch/xpack/stateless/recovery/PointInTimeRelocationIT.java
+++ b/x-pack/plugin/stateless/src/internalClusterTest/java/org/elasticsearch/xpack/stateless/recovery/PointInTimeRelocationIT.java
@@ -765,9 +765,11 @@ public class PointInTimeRelocationIT extends AbstractStatelessPluginIntegTestCas
         ensureGreen(indexName);
         logger.info("Search node " + searchNodeCurrent + " excluded.");
 
-        // PIT id should have changed at some point, and all contexts on the source node should be closed
+        // All contexts on the source node should be closed once the shard has relocated away.
         waitForNoPITContextOnNode(searchNodeCurrent, 5);
-        assertFalse("pit1 should have changed.", isEquivalentId(pitId1.get(), testDataSetup.pitId1));
+        // pitId1 is updated asynchronously by the search threads, so it can briefly still hold the
+        // pre-relocation id even after the source node's contexts are gone; retry until it catches up.
+        assertBusy(() -> assertFalse("pit1 should have changed.", isEquivalentId(pitId1.get(), testDataSetup.pitId1)), 5, TimeUnit.SECONDS);
 
         thread1Running.set(false);
         thread2Running.set(false);

